src/main/java/net/dries007/tfc/common/tileentity/TFCTileEntities.java

```diff
@@ -26,8 +29,24 @@
 {
     public static final DeferredRegister<TileEntityType<?>> TILE_ENTITIES = DeferredRegister.create(ForgeRegistries.TILE_ENTITIES, MOD_ID);
 
-    public static final RegistryObject<TileEntityType<FarmlandTileEntity>> FARMLAND = register("farmland", FarmlandTileEntity::new, TFCBlocks.SOIL.get(SoilBlockType.FARMLAND).values());
+    public static final Collection<RegistryObject<Block>> SAPLING_LIST = new ArrayList<>(TFCBlocks.WOODS.size());
+    static
+    {
+        TFCBlocks.WOODS.forEach((key, value) -> SAPLING_LIST.add(value.get(Wood.BlockType.SAPLING)));
+    }
+
+    public static final RegistryObject<TileEntityType<FarmlandTileEntity>> FARMLAND = register("farmland", FarmlandTileEntity::new, TFCBlocks.SOIL.get(SoilBlockType.FARMLAND).values().stream());
     public static final RegistryObject<TileEntityType<SnowPileTileEntity>> SNOW_PILE = register("snow_pile", SnowPileTileEntity::new, TFCBlocks.SNOW_PILE);
+    public static final RegistryObject<TileEntityType<FirepitTileEntity>> FIREPIT = register("firepit", FirepitTileEntity::new, TFCBlocks.FIREPIT);
+    public static final RegistryObject<TileEntityType<GrillTileEntity>> GRILL = register("grill", GrillTileEntity::new, TFCBlocks.GRILL);
+    public static final RegistryObject<TileEntityType<PotTileEntity>> POT = register("pot", PotTileEntity::new, TFCBlocks.POT);
+    public static final RegistryObject<TileEntityType<TickCounterTileEntity>> TICK_COUNTER = register("tick_counter", TickCounterTileEntity::new, Stream.of(SAPLING_LIST, TFCBlocks.FRUIT_TREE_SAPLINGS.values(), TFCBlocks.FRUIT_TREE_GROWING_BRANCHES.values(), Arrays.asList(TFCBlocks.TORCH, TFCBlocks.WALL_TORCH, TFCBlocks.DEAD_BERRY_BUSH, TFCBlocks.DEAD_CANE, TFCBlocks.BANANA_SAPLING)).flatMap(Collection::stream));
```

Comment:
Instead of
`Stream.of(...).flatMap(Collection::stream)`
In cases where you have to make calls to `Arrays.asList` or `Collections.singletonList` which look really ungangly in stream calls, you can either 1. extract the single elements out of the `flatMap` call, or use `Stream.concat` to join individual elements into the stream. Or stream the elements and ignore the flat map completely.

Reply:
I tried for about 20 minutes to figure out what you meant by this and did not get anywhere, so i will have to leave it for now